api/access.go

```diff
@@ -102,30 +103,20 @@ func Login(username, password string) (sessionID string, user *AuthIDOutput, err
 		panic("Could not marshal auth struct that we JUST made")
 	}
 
-	req, err := http.NewRequest("POST", uri.String(), bytes.NewReader(j))
+	var req *http.Request
+	req, err = http.NewRequest("POST", uri.String(), bytes.NewReader(j))
 	if err != nil {
 		return
 	}
 
-	r, err := makeRequest(req)
+	user = &AuthIDOutput{}
+	var header http.Header
+	header, err = uri.HeaderRequest(user, req)
```

Comment:
Rename `HeaderRequest` to `RequestWithHeaders`

Reply:
Done if the PR ever takes the commit